File: Real-estate-market-IB190099.Services/Database/Property.cs
using System;
using System.Collections.Generic;

namespace Real_estate_market_IB190099.Services.Database;

public partial class Property
{
    public int Id { get; set; }

    public string? Name { get; set; }

    public string? Description { get; set; }

    public int? Floors { get; set; }

    public string? PropertyType { get; set; }

    public int? Rooms { get; set; }

    public byte? Parking { get; set; }

    public byte? Water { get; set; }

    public byte? Electricity { get; set; }

    public int? Price { get; set; }

    public int? AddressId { get; set; }

    public double? Quadrature { get; set; }

    public int? LocationId { get; set; }

    public int? YearOfConstruction { get; set; }

    public virtual Address? Address { get; set; }

    public virtual ICollection<Advertise> Advertises { get; set; } = new List<Advertise>();

    public virtual ICollection<Image> Images { get; set; } = new List<Image>();

    public virtual Location? Location { get; set; }

    public virtual ICollection<Rating> Ratings { get; set; } = new List<Rating>();
}

File: Real-estate-market-IB190099.mobile/realestatemobile/lib/model/property.dart
// ignore_for_file: non_constant_identifier_names, depend_on_referenced_packages
import 'dart:convert';

import 'package:json_annotation/json_annotation.dart';
import 'package:realestatemobile/model/image.dart';
import 'package:realestatemobile/model/location.dart';

import 'address.dart';

part 'property.g.dart';

@JsonSerializable()
class Property {
  Property() {}

  int? id;
  String? name;
  String? description;
  int? floors;
  String? propertyType;
  int? rooms;
  int? yearOfConstruction;
  int? parking;
  int? water;
  int? electricity;
  int? price;
  int? addressId;
  double? quadrature;
  int? locationId;
  List<String>? images;
  Address? address;
  Location? location;

  factory Property.fromJson(Map<String, dynamic> json) =>
      _$PropertyFromJson(json);

  /// Connect the generated [_$PersonToJson] function to the `toJson` method.
  Map<String, dynamic> toJson() => _$PropertyToJson(this);
}
